back/app/models/phase.rb

```diff
@@ -114,6 +119,17 @@ def sanitize_description_multiloc
     self.description_multiloc = service.linkify_multiloc(description_multiloc)
   end
 
+  def validate_campaigns_settings_keys_and_values
+    return if campaigns_settings.blank?
+
+    campaigns_settings.each do |key, value|
+      errors.add(:campaigns_settings, :invalid_key, message: 'invalid key') unless CAMPAIGNS.include?(key.to_sym)
+      next if Utils.boolean? value
+
+      errors.add(:campaigns_settings, :invalid_value, message: 'invalid value')
+    end
+  end
```

Comment:
I designed this to be ready for when/if we add more campaigns to `campaigns_settings`. Maybe unnecessary and we could just hardcode the validation of the specific `project_phase_started` `campaigns_setting` value.

Using a `Utils` method to get around the Rubocop linting warning:
```
Performance/CollectionLiteralInLoop: Avoid immutable Array literals in loops. It is better to extract it into a ***** variable or a constant.
```